k9mail-library/src/test/java/com/fsck/k9/mail/filter/LineWrapOutputStreamTest.java

```diff
@@ -0,0 +1,308 @@
+package com.fsck.k9.mail.filter;
+
+import org.junit.Test;
+
+import java.io.BufferedWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Created by ployt0 on 2/15/17.
+ * Java quotes by Bruce Eckel, Thinking In Java
+ */
+public class LineWrapOutputStreamTest {
+    private ByteArrayOutputStream baos = new ByteArrayOutputStream(64000);
+    private Random rand = new Random();
```

Comment:
I think fuzzy testing has it's uses, especially if, as you say, I record the pseudo random state/seed. The dictionary size in UTF is too vast to test explicitly. Granted I haven't targeted known weaknesses explicitly and eloquently but we don't know what we don't know.
In this case I'm attempting to generate some of the single bytes the filter is looking for, only out of the connect of US ASCII and instead as part of a multi byte char. Randomness is great for pushing internal state, IMHO.